examples/python/async_streaming/server.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2020 The gRPC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import time
+from concurrent.futures import ThreadPoolExecutor
+from typing import Iterable
+
+import grpc
+from google.protobuf.json_format import MessageToJson
+
+import phone_pb2
+import phone_pb2_grpc
+
+
+def create_state_response(call_state: phone_pb2.CallState.State
+                         ) -> phone_pb2.StreamCallResponse:
+    response = phone_pb2.StreamCallResponse()
+    response.call_state.state = call_state
+    return response
+
+
+class Phone(phone_pb2_grpc.PhoneServicer):
+
+    def __init__(self):
+        self._id_counter = 0
+
+    def _create_call_session(self) -> phone_pb2.CallInfo:
+        call_info = phone_pb2.CallInfo()
+        call_info.session_id = str(self._id_counter)
+        self._id_counter += 1
```

Comment:
This is not threadsafe. [It resolves to multiple Python opcodes.](https://godbolt.org/z/9vxYjq)

Reply:
Good catch! TIL.